posthog/hogql/transforms/property_types.py

```diff
@@ -94,7 +94,13 @@ def __init__(
     def visit_field(self, node: ast.Field):
         if isinstance(node.type, ast.FieldType):
             if isinstance(node.type.resolve_database_field(), DateTimeDatabaseField):
-                return ast.Call(name="toTimeZone", args=[node, ast.Constant(value=self.timezone)])
+                return ast.Call(
+                    name="toTimeZone",
+                    args=[node, ast.Constant(value=self.timezone)],
+                    type=ast.CallType(
+                        name="toTimeZone", arg_types=[ast.DateTimeType()], return_type=ast.DateTimeType()
+                    ),
+                )
```

Comment:
Do we need to add the same for `DateDatabaseField`'s, so that data warehouse tables with dates work?

Reply:
This already works for `DateDatabaseField`s (but thanks for calling out, added a test to ensure this), so yeah, if a field is marked as that, this will work.